### PR TITLE
[Instrumentation.Runtime] Update readme after inclusion of process.runtime.dotnet.gc.objects.size

### PR DESCRIPTION
**Pull request by lachmatt**
## Changes

Requested in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/683#discussion_r1023079756

Update `README.md` to reflect addition of `process.runtime.dotnet.gc.objects.size` metric.

Key distinctions between `process.runtime.dotnet.gc.objects.size` and other heap size metrics as identified [here](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/683#discussion_r1010985509) included in metric description.

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 1]
```diff
@@ -63,6 +63,22 @@ The API used to retrieve the value is:
...
+collected yet.
+Fragmentation and other GC committed memory pools are excluded.
+
+| Units   | Instrument Type         | Value Type | Attribute Key(s)  | Attribute Values |
+|---------|-------------------------|------------|-------------------|------------------|
+| `bytes` | ObservableUpDownCounter | `Int64`    | No Attributes     | N/A              |
+
+The API used to retrieve the value is:
+
+* [GC.GetTotalMemory](https://docs.microsoft.com/dotnet/api/system.gc.gettotalmemory):
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:78]: Mention that the `false` is passed to the API?

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 2]
```diff
@@ -63,6 +63,22 @@ The API used to retrieve the value is:
...
+  Retrieves the number of bytes currently thought to be allocated.
+The value is an approximate count.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:80]: would be nice to indicate that this value is not the snapshot from the last GC event, and also that this metric will be available irrespective of if a GC event has occured or not.

File: src/OpenTelemetry.Instrumentation.Runtime/README.md (hunk 1 above)
Comment (lachmatt) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:79]: added in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/772/commits/bd9b48768a89ef6c4d0c055fb85dd2a2898898a3

File: src/OpenTelemetry.Instrumentation.Runtime/README.md (hunk 2 above)
Comment (lachmatt) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md]: added in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/772/commits/bd9b48768a89ef6c4d0c055fb85dd2a2898898a3

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 3]
```diff
@@ -63,6 +63,25 @@ The API used to retrieve the value is:
   The number of times garbage collection has occurred for the specified generation
 of objects.
 
+#### process.runtime.dotnet.**gc.objects.size**
+
+Count of bytes currently in use by objects in the GC heap that haven't been
+collected yet.
+Fragmentation and other GC committed memory pools are excluded.
+The value is available irrespective of GC events.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:71]: Not very clear statement.

Comment (lachmatt) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md]: Updated again in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/772/commits/12919a61ac716f38e7895c0e3d8304895e1839be. 
I tried to be consistent with other metrics description.
Indication that value is not a snapshot is included in description of API used.

**Review by utpilla — APPROVED** [review]

**Pull request merged**